### PR TITLE
Use wheels instead for custom bdists

### PR DESCRIPTION
**Pull request by soltanmm**
cc @nathanielmanistaatgoogle

### CODE REVIEW

**Comment by soltanmm** [comment]
Note that this now requires that a user have installed `wheel`, even if they're downloading a source package.

This... should always be the case?

It doesn't solve some of the dirty, dirty, dirty aspects of name compatibility (see `TODO` comment in commit), but... we have a graceful fallback for that so we can sort of just support people who need binary distributions one at a time on a need-by-need basis (for now).

**Comment by soltanmm** [comment]
Also super, super kudos to @hsaliak.

File: src/python/grpcio/commands.py [hunk 1]
```diff
@@ -74,46 +77,52 @@ class CommandError(Exception):
...
-def _get_grpc_custom_bdist_egg(decorated_basename, target_egg_basename):
-  """Returns a string path to a .egg file for Linux to install.
+def _get_grpc_custom_bdist(decorated_basename, target_bdist_basename):
+  """Returns a string path to a bdist file for Linux to install.
 
-  If we can retrieve a pre-compiled egg from online, uses it. Else, emits a
+  If we can retrieve a pre-compiled bdist from online, uses it. Else, emits a
   warning and builds from source.
   """
+  # TODO(atash): somehow the name that's returned from `wheel` is different
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/commands.py]: I don't understand this. Is something wrong with `wheel`? If so, link to a wheel bug report? Is `wheel` missing something that you feel we need? If so, link to a wheel feature request? If neither, I'm really unclear on what work is to be done here that we need TODO but are not able to do at this time?

File: src/python/grpcio/commands.py [hunk 2]
```diff
@@ -74,46 +77,52 @@ class CommandError(Exception):
...
-        '{}\n\nCould not write grpcio egg: {}'
+        '{}\n\nCould not write grpcio bdist: {}'
             .format(traceback.format_exc(), error.message))
-  return egg_path
+  return bdist_path
 
 
-class EggNameMixin(object):
+class WheelNameMixin(object):
   """Mixin for setuptools.Command classes to enable acquiring the egg name."""
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/commands.py]: "egg name" on this line feels stale. Should the word "egg" even appear anywhere in this file any more?

File: src/python/grpcio/commands.py (hunk 1 above)
Comment (soltanmm) [inline src/python/grpcio/commands.py]: The TODO is to replicate the compatibility check `wheel` does. We don't have the luxury of doing that at this time.

**Pull request merged**